### DISCUSSION TITLE
TourIonPopoverModule and Custom Template problem

### DISCUSSION THREAD

**Discussion by NSiggel** [hakimio/ngx-ui-tour#157] [Q&A] [answered]
Currently trying to use 
`  <tour-step-template>
    <ng-template let-step="step">`
...

however even with no customization, using de default from website, the templates shows up non clickable state, Frozen on screen, and can't be dismissed.
Visually I can also see that the CSS does not line up the way the default one with no customization does
<tour-step-template></tour-step-template>

Does anyone have proper guidance on setting this up in an TourIonPopoverModule context ?
I've got my <tour-step-template> at the bottom of my 
app.component.html
and I've registers a number of steps and achors and using default template everything works, but feel there might be something missing in docs of configuration in order to be able to use the ng-template properly. 

The primary reason for needing the template is the ability to embed a iframe with video content.
This does not work, as the Angular XSS it is currently stripping the content out. 

I know we can fix this by using domsanitizer which I would apply via a pipe, but I feel I would need to use the tour-step-template in order to apply something like
            <ion-card-content
                [innerHTML]="step.content | noSanitizing"
            ></ion-card-content>
      
Any suggestion would be welcome.

**Comment by hakimio** [comment]
Can you create reproduction of your "non-clickable" tour? What version of the tour are you using?

**Reply by hakimio** [reply]
You can find default step template [here](https://github.com/hakimio/ngx-ui-tour/blob/master/projects/ngx-ui-tour-ionic/src/lib/tour-step-template/tour-step-template.component.ts).

**Reply by hakimio** [reply, marked as answer]
CSS [here](https://github.com/hakimio/ngx-ui-tour/blob/master/projects/ngx-ui-tour-ionic/src/lib/tour-step-template/tour-step-template.component.scss)

**Comment by NSiggel** [comment]
Let me see about creating a reproduction.
As to core versions
"@angular/core": "^15.0.0",
"@ionic/angular": "^7.1.0",
"ngx-ui-tour-ionic": "^1.7.0",

when I copy paste the default template in from
https://hakimio.github.io/ngx-ui-tour/ion-popover/Misc#custom-template

this is a screenshot I get. Notice the X button for close is not in the right spot.
<img width="482" alt="image" src="https://github.com/hakimio/ngx-ui-tour/assets/11523121/d5ca49f0-0572-4203-95a9-56655a030453">

with the step definition as
{
          anchorId: 'start-tour',
          title: 'Welcome To Tour',
          content: `Welcome to the Ngx-UI-Tour tour!<br><br>Let's begin!`,
          enableBackdrop: true,
          stepDimensions: {
            minWidth: '500px',
          },
        },

So it's as though the template did not register itself properly.
we can see the step min-width is being applied. the the content of the template is not being inserted/registered properly.

- From the new link, should I be implementing it the full class as my own component ?
- is this a possible version compatibility issue ?

really appreciate your quick replies and assistance.

**Reply by hakimio** [reply]
The default template in the docs is for the new version: v2

**Reply by NSiggel** [reply]
Thanks for confirming.
So maybe the key question does ionic 1.7 allow for a custom <tour-step-template> ?
Does it need to be registered in a different way ? vs docs which seem to be on the current release ?

I tested the following template in the 1.7 branch, but getting the same errors.
https://github.com/hakimio/ngx-ui-tour/blob/ionic%40v1.7.0/src/app/ion-popover/examples/default-template.txt

Do you have any other suggestions on getting iframe or remote content in the content of a step ?
I'm assuming the [innerHTML]=step.content with a sanitizer is the best way.

Really appreciate any insight. Will look at v2 migration.

**Reply by hakimio** [reply]
> So maybe the key question does ionic 1.7 allow for a custom ?

Custom templates work the same way. Just default template CSS is slightly different.

Anyway, did you manage to make your custom template with `iframe` work?
